# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/hassan9039/intern1/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

My provisional lane: Lane 2 — Refresh / Content Opportunity Scoring.

I want to investigate which pages should be reviewed first for a content refresh or other SEO action. This lane interests me because the starter data already contains signals such as CTR, average position, impressions, content age, and trend direction. These signals may help prioritize pages for human review instead of treating every page equally. I am choosing this as a provisional lane because the evidence from the starter dataset suggests there may be useful patterns, but I will validate the question further as I work with more data.

In [3]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

Search question: Which pages should be prioritized for human review for a content refresh or other SEO action?

Decision: Decide which pages are worth reviewing first rather than treating all pages equally.

Action: An SEO/content team could review the highest-priority pages and decide whether to refresh, expand, protect, or otherwise improve them.

Cost of a wrong call: A false positive could waste time and resources on a page that did not need attention. A false negative could cause a potentially valuable page to be overlooked, resulting in a missed opportunity to improve its performance.

In [4]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [5]:
import pandas as pd

url = "https://raw.githubusercontent.com/hassan9039/intern1/main/data/raw/content_refresh_anonymized.csv"

df = pd.read_csv(url)

print("Total pages:", len(df))
print("Mean CTR:", round(df["ctr"].mean(), 3))
print("Mean impressions (90d):", round(df["impressions_90d"].mean(), 1))

Total pages: 30000
Mean CTR: 0.511
Mean impressions (90d): 5200.4


The starter dataset contains 30,000 pages, giving enough observations to investigate page-level prioritization. The mean CTR is 0.511, while pages receive an average of about 5,200 impressions over 90 days. These numbers suggest that the dataset contains substantial search activity and measurable page-performance signals that could be useful for prioritizing pages for human review. This does not yet show that a refresh will improve performance, but it gives a reasonable basis for investigating which pages may deserve attention.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

What I can claim: The starter data shows differences in page-level performance metrics such as CTR, impressions, average position, content age, and trend direction. These signals can be used to investigate whether some pages appear more suitable for prioritization than others.

What I cannot claim yet: I cannot claim that refreshing a page will cause its CTR, rankings, or traffic to improve. I also cannot claim that the patterns in this starter dataset will generalize to every client or website. Any model or scoring system should support human decision-making rather than automatically deciding what action to take.

In [6]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.